pyquil/api/qpu.py

```diff
@@ -158,23 +167,26 @@ def run_and_measure(self, quil_program, qubits, trials=1):
         :param Program quil_program: Pyquil program to run on the QPU
         :param list|range qubits: The list of qubits to measure
         :param int trials: Number of times to run the program (a.k.a. number of shots)
+        :param bool needs_compilation: If True, preprocesses the job with the compiler.
+        :param ISA isa: If set, specifies a custom ISA to compile to. If left unset,
+                    Forest uses the default ISA associated to this QPU device.
         :return: A list of a list of classical registers (each register contains a bit)
         :rtype: list
         """
-        job = self.wait_for_job(self.run_and_measure_async(quil_program, qubits, trials))
+        job = self.wait_for_job(self.run_and_measure_async(quil_program, qubits, trials, needs_compilation, isa))
         return job.result()
 
-    def run_and_measure_async(self, quil_program, qubits, trials):
+    def run_and_measure_async(self, quil_program, qubits, trials, needs_compilation=True, isa=None):
         """
         Similar to run_and_measure except that it returns a job id and doesn't wait for the program
         to be executed. See https://go.rigetti.com/connections for reasons to use this method.
         """
         full_program = append_measures_to_program(quil_program, qubits)
-        payload = self._run_and_measure_payload(full_program, qubits, trials)
+        payload = self._run_and_measure_payload(quil_program, qubits, trials, needs_compilation=needs_compilation, isa=None)
```

Comment:
isa=None?

Reply:
I swear that you'd already found this and I'd already fixed it.